test/dump.ts

```diff
@@ -74,4 +74,10 @@ describe('dump*()', () => {
         writer.close();
         assert.deepEqual(ionBinary, writer.getBytes());
     });
+    it('Struct with annotated field large enough to require a VarUInt length', () => {
```

Comment:
Could you also add the example input from issue #621 as a separate unit test? Add a comment with the GitHub issue link too, please.